packages/bpk-component-badge/src/BpkBadge.module.scss

```diff
@@ -18,50 +18,108 @@
 
 @import '../../bpk-mixins/index.scss';
 
-.bpk-badge {
-  @include bpk-badge;
+.base {
+  display: inline-flex;
+  padding: bpk-spacing-sm() bpk-spacing-md();
+  align-items: center;
 
-  &--centered {
-    @include bpk-badge--centered;
-  }
-
-  &--docked-right {
-    @include bpk-badge--docked-right;
-  }
-
-  &--docked-left {
-    @include bpk-badge--docked-left;
-  }
+  @include bpk-border-radius-xs;
+  @include bpk-text;
+  @include bpk-caption;
+}
+.warning {
+  composes: base;
+  background-color: $bpk-status-warning-fill-day;
+  color: $bpk-text-on-light-day;
+  fill: $bpk-text-on-light-day;
+}
+.normal {
+  composes: base;
+  background-color: $bpk-surface-highlight-day;
+  color: $bpk-text-primary-day;
+  fill: $bpk-text-primary-day;
+}
+.success {
+  composes: base;
+  background-color: $bpk-status-success-fill-day;
+  color: $bpk-text-on-light-day;
+  fill: $bpk-text-on-light-day;
+}
+.critical {
+  composes: base;
+  background-color: $bpk-status-danger-fill-day;
+  color: $bpk-text-on-light-day;
+  fill: $bpk-text-on-light-day;
+}
+.inverse {
+  composes: base;
+  background-color: $bpk-surface-default-day;
+  color: $bpk-text-primary-day;
+  fill: $bpk-text-primary-day;
+}
+.outline {
+  composes: base;
+  background-color: transparent;
+  color: $bpk-text-on-dark-day;
+  box-shadow: inset 0 0 0 $bpk-border-size-sm $bpk-text-on-dark-day;
+  fill: $bpk-text-on-dark-day;
+}
+.strong {
+  composes: base;
+  background-color: $bpk-core-primary-day;
+  color: $bpk-text-on-dark-day;
+  fill: $bpk-text-on-dark-day;
+}
+.brand {
+  composes: base;
+  background-color: $bpk-core-accent-day;
+  color: $bpk-text-primary-inverse-day;
+  fill: $bpk-text-primary-inverse-day;
+}
 
-  &--warning {
-    @include bpk-badge--warning;
-  }
+$types: 'warning', 'normal', 'success', 'critical', 'inverse', 'outline', 'strong', 'brand';
 
-  &--success {
-    @include bpk-badge--success;
+@each $type in $types {  
```

Comment:
could you explain this please?

Reply:
This is sass basically what it is doing is using some loops to avoid repetitive typing. As we now know, CSS modules demands a single class per element. That makes variants a little tricky, in addition to having `.warning` we'd need `warning-centered` `warning-right` and `warning-left`. We need these for _every_ "type" of badge. That would be a lot of typing out the same thing. So instead we can write a loop. 

`$types: 'warning', 'normal', 'success', 'critical', 'inverse', 'outline', 'strong', 'brand';`

is equivalent to `const types = ['warning', 'normal', 'success', 'critical', 'inverse', 'outline', 'strong', 'brand']`

`@each $type in $types`

is equivalent to `for type in types`

Now we have a loop with all our types & it's time to consume that information: 

```
  .#{$type}-centered {
    composes: #{$type};
    vertical-align: text-bottom;
  }
```

`#{$type}` is the syntax for string interpolation, it's equivalent to: `'${type}'`. We are creating a new class, when `type=warning` we get: `warning-centered` & we are doing it for each "type"!.

This way, we can apply a _single_ style to each element that entirely defines it's behaviour. 

This notably doesn't allow for _multiple_ variants to be applied at once. Eg you can't be `right` and `centered`. We could support that, I've not done that in this PR though.